### DISCUSSION TITLE
input_datetime attribute timestamp

### DISCUSSION THREAD

**Discussion by Petro31** [home-assistant/architecture#523] [Entity Models]
Time is a royal pain for most people in home assistant, however we've made plenty of ways to get around using complex calculations.  One thing I've noticed when dealing with support is the timestamp attribute is pretty useless for input_datetime 
state objects.

1.  It assumes UTC, not local.
2.  If you don't include the date, it doesn't include the date in the timestamp.  Even when it does include the date, it treats the time as UTC and not local making simple time comparisons useless in templates.

Is there a reason we decided this?  It's painful to use in templates.  For example if you want to use relative time, you need to convert it to a datetime object.  But you have to add the date and the time zone.  The time zone you get from timestamp_custom is not the correct format and cannot be used in strptime.  So you have to insert a colon before the 4th character then concatenate the date and the time zone to get the proper datetime.  This is just as difficult to use when managing them as timestamp integers.  

The outcome to this in my opinion is 1 of 2 options:

1.  Change this to a timestamp including the date and time zone.
2.  Add a new datetime attribute to these state objects that contains the date and time zone.

This way, users can use these for easy date calculations.  Or they can simply drop it into a device_class timestamp sensor if they want to display the relative time in the UI.

**Comment by koying** [comment]
I looked at implementing a full timestamp for "time-only" input_datetime in https://community.home-assistant.io/t/trigger-automation-15-minutes-before-date-time-stored-in-input-datetime/145279/40?u=koying , but as Taras correctly remarked, adding the date (of the day?) to make up a full datetime implies that `input_datetime` be regularly updated, while it is not (I assume not input_* are, really).
The other 2 variants, full datetime and date-only actually return an actual UNIX epoch in "timestamp".

That basically only leaves the filter option, as you mention in that thread.
- Or an ad-hoc `today` filter that would specifically create a full datetime from today's date and the input_datetime
- Or, more generically, more-or-less expose the `combine` method of python datetime, that takes a date and a time to make up a full datetime.

Something like
```
to_datetime(now(), state('input_datetime.time_only'))
```

**Reply by koying** [reply]
Or, maybe easier, just a `seconds_since_midnight` template function, that would return the time of day in seconds, for easy comparison to the timestamp attribute of a time-only input_datetime?

**Reply by koying** [reply]
Or modify `as_datetime`/`as_timestamp` to handle that special case. Probably the straightforward way

**Comment by Petro31** [comment]
I don't think this is relevant anymore.
